CHANGES.md

```diff
@@ -8,6 +8,50 @@
 - port to conduit 3.0.0: minor breaking changes on the server API, an explicit distinction
   between cohttp-lwt-unix (using tls), cohttp-lwt-notls and cohttp-lwt-ssl (using lwt_ssl),
   and includes ssl in cohttp-async (@dinosaure #692)
+
+  **breaking changes**, the API to launch a server was updated and types used by the client
+  were updated too. A clean-recompilation of your project should be enough for users of the
+  client part - however, if you deeply use Conduit, you should look the release of Conduit
+  3.0.0 (documentation & howto) to be aware about the new usage of this library.
+
+  [Release of Conduit 3.0.0](https://github.com/mirage/ocaml-conduit/releases/tag/v3.0.0)
+  [HOW-TO use Conduit 3.0.0](https://mirage.github.io/ocaml-conduit/conduit/howto.html)
+  [Documentation of Conduit 3.0.0](https://mirage.github.io/ocaml-conduit/conduit/Conduit/module-type-S/index.html)
+
+  About the server-side part of Cohttp, we broke the API according to Conduit 3.0.0. The user
+  must update the way to launch a Cohttp server. The technical update is about the `~mode`
+  argument which disappears to let 3 arguments: `cfg`, `service` and `protocol`.
+
+  They permit to the user to choose which kind of service he wants (a TLS service or
+  a simple TCP service). Such values are provided by Conduit 3.0.0. For example, if
+  you want to launch a simple HTTP (no secure) service, you can use:
+  - `Conduit_{async,lwt}.TCP.service`
+  - `Conduit_{async,lwt}.TCP.protocol`
+
+  The encryption layer can be provided by `Conduit_{async,lwt}_{tls,ssl}.TCP` module (`tls` means
+  an usage of `ocaml-tls`, `ssl` means a usage of OpenSSL). Then, the `cfg` value depends on the
+  `service` value. For example, for `Conduit_lwt.TCP.service`, `cfg = Conduit_lwt.TCP.configuration`.
+  On this way, the user is able to launch a HTTP server with:
+
+```ocaml
+let cfg =
+  { Conduit_lwt.TCP.sockaddr= Unix.ADDR_INET (Unix.inet_addr_loopback, 8080)
+  ; capacity= 40 }
+
+let run cohttp_config =
+  Cohttp_lwt_unix.Server.create cfg Conduit_lwt.TCP.protocol Conduit_lwt.TCP.service
```

Comment:
out of curiosity when do weed need to both specify a protocol and a service? I though the protocol was included in the service?

Reply:
It's needed to explain how to _pack_ the given flow. The current implementation of `Conduit.serve_with_handler` gives the real type/value. Then, `Cohttp` packs this value to the abstract `Conduit.flow` value (but it needs to know the _protocol_ to do this operation).

Reply:
that part of the API is still a bit confusing to me ; I'm wondering if we could make this simpler by having to somehow pass only the service (for instance by extending a bit the definition of what a service is).

Reply:
(note: I am fine with the current API, I'm just suggesting possible improvement for conduit 3.1)

Reply:
We still have time to change it if there is an agreement, and avoid another breaking ( I would call it 4.0.0) release :P

Reply:
Agreed, I find this bit confusing too. Now's the time to tweak it, not 3.1 :-)

Reply:
Lol I had just merged the PR. I can remove the announcement